### PR TITLE
[ADD] asset preference and item coloring permissions

### PR DESCRIPTION
**Pull request by Sekkmer**

### CODE REVIEW

File: pandora-common/src/character/restrictionsManager.ts [hunk 1]
```diff
@@ -296,10 +297,17 @@ export class CharacterRestrictionsManager {
...
-				case 'favorite':
-				case 'normal':
 				case 'maybe':
+					context.addInteraction(target.character, 'assetPrefMaybe');
+				// Fallthrough
+				case 'normal':
+					context.addInteraction(target.character, 'assetPrefNormal');
+				// Fallthrough
+				case 'favorite':
+					context.addInteraction(target.character, 'assetPrefFavorite');
```
Comment (Jomshir98) [inline pandora-common/src/character/restrictionsManager.ts:307]: Should they really be fallthrough? I'm not fully sure~

File: pandora-common/src/gameLogic/interactions/_interactionConfig.ts [hunk 2]
```diff
@@ -36,12 +42,30 @@ export const INTERACTION_CONFIG = {
...
+		defaultPermissions: {
+			allowOthers: true,
+		},
+	},
+	assetPrefMaybe: {
+		visibleName: 'Add and remove assets with maybe preference',
+		defaultPermissions: {
+			allowOthers: true,
+		},
+	},
```
Comment (Jomshir98) [inline pandora-common/src/gameLogic/interactions/_interactionConfig.ts:62]: I'm not sure about these being part of interactions config... I would rather move them to be their own category. If you wouldn't mind I can implement that as I have a good idea how to do that. (I intended for the solution to be used that way, but there is so far no example of how I actually imagined it being done)

File: pandora-common/src/assets/appearanceActions.ts [hunk 3]
```diff
@@ -366,22 +328,14 @@ export function DoAppearanceAction(
...
-			if (!r.allowed) {
-				processingContext.addProblem({
-					result: 'restrictionError',
-					restriction: r.restriction,
-				});
+			processingContext.checkCanUseItem(target, action.item, ItemInteractionType.STYLING);
+
+			if (target.type === 'character') {
+				processingContext.addInteraction(target.character, 'changeItemColor');
 			}
```
Comment (Jomshir98) [inline pandora-common/src/assets/appearanceActions.ts:339]: Thought: Maybe worth moving this interaction check into `canUseItem` when the interaction type is `STYLING`? That type is only used for coloring right now (and in theory for modules that mark themselves as styling, which makes sense for some modules like light/dark variant).

**Review by Jomshir98 — CHANGES_REQUESTED** [review]

File: pandora-common/src/gameLogic/assetPreferences/assetPreferencesSubsystem.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+} as const satisfies Immutable<Record<AssetPreferenceType, AssetPreferencePermissionConfig | null>>;
+
+export interface AssetPreferencePermissionConfig {
+	visibleName: string;
+	defaultPermissions: PermissionConfigDefault;
+}
+
+export abstract class AssetPreferencesSubsystem extends TypedEventEmitter<AssetPreferencesSubsystemEvents> implements IPermissionProvider {
+	public abstract get currentPreferences(): Immutable<AssetPreferencesPublic>;
+	public abstract getPreferencePermission(preference: AssetPreferenceType): GameLogicPermission | null;
```
Comment (Sekkmer) [inline pandora-common/src/gameLogic/assetPreferences/assetPreferencesSubsystem.ts:42]: This feels like a complete duplication of the other Subsystem, but for some reason, this function can return null, the other throws an exception.

What exactly do we gain from this much extra duplicated code, if we just want to split it for the UI, a simple group in the config would have been able to accomplish the same thing with much less code

Comment (Sekkmer) [reply pandora-common/src/gameLogic/assetPreferences/assetPreferencesSubsystem.ts:42]: well I tried this but over all it did not make anything better, the duplication should be fine for now, I guess

**Review by Jomshir98 — APPROVED** [review]

**Pull request merged**